File: app/src/main/java/com/usharik/app/UiLanguage.java
package com.usharik.app;

import android.content.Context;

public enum UiLanguage {
    SYSTEM("system", ""),
    ENGLISH("en", "English"),
    RUSSIAN("ru-RU", "Русский"),
    CZECH("cs", "Čeština"),
    BELARUSIAN("be", "Беларуская"),
    UKRAINIAN("uk-UA", "Українська"),
    GERMAN("de", "Deutsch"),
    VIETNAMESE("vi", "Tiếng Việt");

    private final String preferenceValue;
    private final String displayName;

    UiLanguage(String preferenceValue, String displayName) {
        this.preferenceValue = preferenceValue;
        this.displayName = displayName;
    }

    public String preferenceValue() {
        return preferenceValue;
    }

    public String languageTags() {
        return SYSTEM == this ? "" : preferenceValue;
    }

    public String displayName(Context context) {
        return SYSTEM == this ? context.getString(R.string.use_device_language) : displayName;
    }

    public static UiLanguage fromPreference(String value) {
        if (value != null) {
            for (UiLanguage uiLanguage : values()) {
                if (uiLanguage.preferenceValue.equalsIgnoreCase(value)) {
                    return uiLanguage;
                }
            }
        }
        return SYSTEM;
    }
}

File: app/src/main/kotlin/com/usharik/app/UiLanguage.kt
package com.usharik.app

import android.content.Context

enum class UiLanguage(private val preference: String, private val label: String) {
    SYSTEM("system", ""), ENGLISH("en", "English"), RUSSIAN("ru-RU", "Русский"),
    CZECH("cs", "Čeština"), BELARUSIAN("be", "Беларуская"), UKRAINIAN("uk-UA", "Українська"),
    GERMAN("de", "Deutsch"), VIETNAMESE("vi", "Tiếng Việt");

    fun preferenceValue() = preference
    fun languageTags() = if (this == SYSTEM) "" else preference
    fun displayName(context: Context) = if (this == SYSTEM) context.getString(R.string.use_device_language) else label
    companion object {
        @JvmStatic fun fromPreference(value: String?) = entries.firstOrNull { it.preference.equals(value, true) } ?: SYSTEM
    }
}
